src/cmd/services/m3query/config/config.go

```diff
@@ -94,10 +101,47 @@ type Configuration struct {
 	// Ingest is the ingest server.
 	Ingest *IngestConfiguration `yaml:"ingest"`
 
+	// Carbon is the carbon configuration.
+	Carbon *CarbonConfiguration `yaml:"carbon"`
+
 	// Limits specifies limits on per-query resource usage.
 	Limits LimitsConfiguration `yaml:"limits"`
 }
 
+// CarbonConfiguration returns the carbon configuration.
+func (c *Configuration) CarbonConfiguration() *CarbonConfiguration {
```

Comment:
Can we make this return `CarbonConfiguration` instead of `*CarbonConfiguration`?  That way it's always not nil when used as a local var returned from this method.

Reply:
ended up not needing this accessor at all. Made everything non-point and then used the getter pattern you described above.